pkg/client/typed/scale/scale.go

```
@@ -0,0 +1,218 @@
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/typed/dynamic"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type Client struct {
+	dc *dynamic.Client
+}
+
+type ScaleInterface interface {
+	Replicas() (int, error)
+	SetReplicas(replicas int)
+	Selector() (string, error)
+}
+
+func NewClient(conf *restclient.Config) (*Client, error) {
+	// Make a copy, do not modify the passed config.
+	// TODO(madhusudancs): Should we just receive the conf argument by value? Receiving this as a
+	// pointer now just to be consistent with other clients.
+	conf2 := *conf
+	if conf2.APIPath == "" {
+		conf2.APIPath = "/apis"
+	}
+
+	dc, err := dynamic.NewClient(&conf2)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		dc: dc,
+	}, nil
+}
+
+// NewForConfigOrDie creates a new dynamic client for the given config and
+// panics if there is an error in the config.
+func NewForConfigOrDie(conf *restclient.Config) *Client {
+	cl, err := NewClient(conf)
+	if err != nil {
+		panic(err)
+	}
+	return cl
+}
+
+// TODO(madhusudancs): Implement this as a codec decoder.
+func (c *Client) decode(res *runtime.Unstructured) (ScaleInterface, error) {
+	if res.TypeMeta.Kind != "Scale" {
+		return nil, fmt.Errorf("object should be of Kind `Scale`, got `%s`", res.TypeMeta.Kind)
+	}
+	switch res.TypeMeta.APIVersion {
+	case "extensions/v1beta1":
+		return &extScale{scale{c, res}}, nil
+	case "autoscaling/v1":
+		return &asScale{scale{c, res}}, nil
+	default:
+		return nil, fmt.Errorf("unknown APIVersion `%s` for `Scale`", res.TypeMeta.APIVersion)
+	}
+}
+
+func (c *Client) Get(scaleRef extensions.SubresourceReference, namespace string) (ScaleInterface, error) {
+	gv, err := unversioned.ParseGroupVersion(scaleRef.APIVersion)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing group version from scale reference: %v", err)
+	}
+	gvr, _ := meta.KindToResource(gv.WithKind(scaleRef.Kind))
```

why is it safe to ignore this error?